test/jest/roktManager.spec.ts

```diff
@@ -769,6 +811,187 @@ describe('RoktManager', () => {
+                    getUserIdentities: () => ({
+                        userIdentities: {
+                            email: 'same@example.com'
+                        }
+                    }),
+                    setUserAttributes: jest.fn()
+                }),
+                identify: jest.fn()
+            };
+
+            roktManager['identityService'] = mockIdentity as unknown as SDKIdentityApi;
+
+            const options: IRoktSelectPlacementsOptions = {
+                attributes: {
+                    email: 'same@example.com'
+                }
+            };
+
+            roktManager.selectPlacements(options);
+
+            expect(mockIdentity.identify).not.toHaveBeenCalled();
+            expect(mockMPInstance.Logger.warning).not.toHaveBeenCalled();
+        });
+
+        it('should call identify with new email when current user has no email', async () => {
+            const kit: Partial<IRoktKit> = {
+                launcher: {
+                    selectPlacements: jest.fn(),
+                    hashAttributes: jest.fn()
+                },
+                selectPlacements: jest.fn(),
+                hashAttributes: jest.fn(),
+                setExtensionData: jest.fn(),
+            };
+
+            roktManager.kit = kit as IRoktKit;
+
+            const mockIdentity = {
+                getCurrentUser: jest.fn().mockReturnValue({
+                    getUserIdentities: () => ({
+                        userIdentities: {}
+                    }),
+                    setUserAttributes: jest.fn()
+                }),
+                identify: jest.fn((data, callback) => {
+                    if (callback) {
+                        callback();
+                    }
+                })
+            } as unknown as SDKIdentityApi;
+
+            roktManager['identityService'] = mockIdentity;
+
+            const options: IRoktSelectPlacementsOptions = {
+                attributes: {
+                    email: 'new@example.com'
+                }
+            };
+
+            await roktManager.selectPlacements(options);
+
+            expect(mockIdentity.identify).toHaveBeenCalledWith({
+                userIdentities: {
+                    email: 'new@example.com'
+                }
+            }, expect.any(Function));
+            expect(mockMPInstance.Logger.warning).not.toHaveBeenCalled();
+        });
+
+        it('should handle identify failure and log error', async () => {
+            const kit: Partial<IRoktKit> = {
+                launcher: {
+                    selectPlacements: jest.fn(),
+                    hashAttributes: jest.fn()
+                },
+                selectPlacements: jest.fn().mockResolvedValue({}),
+                hashAttributes: jest.fn(),
+                setExtensionData: jest.fn(),
+            };
+
+            roktManager.kit = kit as IRoktKit;
+
+            const mockError = { error: 'Identity service error', code: 500 };
```

Comment:
I think it might be more likely to encounter 401 or 403, so we should also test for those.